### PR TITLE
Bump golang 1.20.3 and golint changes

### PR DESCRIPTION
**Pull request by sethiyash**
- Bump golang to 1.20.3 to fix CVE's in go 1.19.6

### CODE REVIEW

File: pkg/vendir/fetch/ref_fetcher.go [hunk 1]
```diff
@@ -27,6 +27,6 @@ func (f SingleSecretRefFetcher) GetSecret(name string) (ctlconf.Secret, error) {
 	return ctlconf.Secret{}, fmt.Errorf("Not found")
 }
 
-func (f SingleSecretRefFetcher) GetConfigMap(name string) (ctlconf.ConfigMap, error) {
+func (f SingleSecretRefFetcher) GetConfigMap(_ string) (ctlconf.ConfigMap, error) {
```
Comment (kumaritanushree) [inline pkg/vendir/fetch/ref_fetcher.go:30]: If we are not using this param here then why we are not removing it instead of using `_`?

Comment (sethiyash) [reply pkg/vendir/fetch/ref_fetcher.go:30]: Here `GetConfigMap` is part of `RefFetcher` interface and we can't change the method signatures.

File: pkg/vendir/directory/staging_dir.go [hunk 2]
```diff
@@ -104,7 +104,7 @@ func (d StagingDir) CopyExistingFiles(rootDir string, stagingPath string, ignore
 		}
 
 		// Move the file to the staging directory
-		err = copy(path, stagingPath)
+		err = copyFunc(path, stagingPath)
```
Comment (praveenrewar) [inline pkg/vendir/directory/staging_dir.go]: Is there a copy function introduced in Go v1.20? 
nit: copyFunc -> copyFile

Comment (sethiyash) [reply pkg/vendir/directory/staging_dir.go]: Not sure when it got introduced but this is what golint is throwing::
```
pkg/vendir/directory/staging_dir.go:197:1: redefines-builtin-id: redefinition of the built-in function copy (revive)
```

**Review by praveenrewar — APPROVED** [review]
LGTM!

**Pull request merged**